pjlib/include/pj/argparse.h

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright (C) 2008-2024 Teluu Inc. (http://www.teluu.com)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+#ifndef __PJ_ARGPARSE_H__
+#define __PJ_ARGPARSE_H__
+
+/**
+ * @file argparse.h
+ * @brief Command line argument parser
+ */
+#include <pj/ctype.h>
+#include <pj/errno.h>
+#include <pj/string.h>
+
+PJ_BEGIN_DECL
+
+/**
+ * Define function to display parsing error.
+ */
+#ifndef PJ_ARGPARSE_ERROR
+#  include <stdio.h>
+#  define PJ_ARGPARSE_ERROR(fmt, arg) printf(fmt "\n", arg)
+#endif
+
+
+/**
+ * @defgroup PJ_ARGPARSE Command line argument parser
+ * @ingroup PJ_MISC
+ * @{
+ *
+ * This module provides header only utilities to parse command line arguments.
+ * This is mostly used by PJSIP test and sample apps. Note that there is
+ * getopt() implementation in PJLIB-UTIL (but it's in PJLIB-UTIL, so it can't
+ * be used by PJLIB)
+ */
+
+/**
+ * Peek the next possible option from argv. An argument is considered an
+ * option if it starts with "-" and followed by at least another letter that
+ * is not digit or starts with "--" and followed by a letter.
+ *
+ * @param argv      The argv, which must be null terminated.
+ *
+ * @return next option or NULL.
+ */
+PJ_INLINE(char*) pj_argparse_peek_next_option(char *const argv[])
+{
+    while (*argv) {
+        const char *arg = *argv;
+        if ((*arg=='-' && *(arg+1) && !pj_isdigit(*(arg+1))) ||
+            (*arg=='-' && *(arg+1)=='-' && *(arg+2)))
+        {
+            return *argv;
+        }
+        ++argv;
+    }
+    return NULL;
+}
+
+/**
+ * Check that an option exists, without modifying argv.
+ *
+ * @param opt       The option to find, e.g. "-h", "--help"
+ * @param argv      The argv, which must be null terminated.
+ *
+ * @return PJ_TRUE if the option exists, else PJ_FALSE.
+ */
+PJ_INLINE(pj_bool_t) pj_argparse_exists(const char *opt, char *const argv[])
```

Comment:
I could be mistaken here, but usually it's the other way around, `pj_argparse_exists(argv, opt)`, with the object being searched (i.e. the main object) being the first parameter.

Reply:
I think that's a good point. Rearranged the argument (also in `pj_argparse_get_bool()/_str()/_int()` to the same convention) in latest commit